extensions/azdata/src/extension.ts

```diff
@@ -14,126 +14,126 @@ let localAzdata: IAzdataTool | undefined = undefined;
 let eulaAccepted: boolean = false;
 
 export async function activate(context: vscode.ExtensionContext): Promise<azdataExt.IExtension> {
-	localAzdata = await checkForAzdata();
-	eulaAccepted = !!context.globalState.get(constants.acceptEula);
+	vscode.commands.registerCommand('azdata.acceptEula', async () => {
+		eulaAccepted = await promptForEula(context.globalState, true /* userRequested */);
+		await vscode.commands.executeCommand('setContext', 'azdata.eulaAccepted', eulaAccepted);
+	});
+
+	vscode.commands.registerCommand('azdata.install', async () => {
+		await checkAndInstallAzdata(true /* userRequested */);
+	});
+
+	vscode.commands.registerCommand('azdata.upgrade', async () => {
+		if (await checkAndUpgradeAzdata(localAzdata, true /* userRequested */)) { // if an upgrade was performed
+			localAzdata = await findAzdata(); // find and save the currently installed azdata
+		}
+	});
+
+	// Don't block on this since we want the extension to finish activating without needing user input
+	checkAndInstallAzdata() // install if not installed and user wants it.
+		.then(async azdataTool => {
+			localAzdata = azdataTool;
+			if (localAzdata !== undefined) {
+				await vscode.commands.executeCommand('setContext', 'azdata.found', true);
+				try {
+					//update if available and user wants it.
+					if (await checkAndUpgradeAzdata(localAzdata)) { // if an upgrade was performed
+						localAzdata = await findAzdata(); // find and save the currently installed azdata
+					}
+				} catch (err) {
+					vscode.window.showWarningMessage(loc.upgradeError(err));
+				}
+			}
+		});
+
+	eulaAccepted = !!context.globalState.get<boolean>(constants.acceptEula);
+	Logger.log(`eulaAccepted state on startup: ${eulaAccepted}`);
 	if (!eulaAccepted) {
 		// Don't block on this since we want extension to finish activating without requiring user actions.
 		// If EULA has not been accepted then we will check again while executing azdata commands.
 		promptForEula(context.globalState)
-			.then(userResponse => {
+			.then(async (userResponse: boolean) => {
 				eulaAccepted = userResponse;
+				await vscode.commands.executeCommand('setContext', 'azdata.eulaAccepted', true);
 			})
-			.catch(err => console.log(err));
+			.catch((err: any) => console.log(err));
```

Comment:
Not sure why you added this - the typing is unnecessary here (and not useful since it's just any anyways)

Reply:
removed. I had added because of some squiggly in vscode.